plugins/eval.py

```diff
@@ -56,10 +59,40 @@ def code_print(*args, sep=" ", end="\n", file=fp, flush=False):
         mk_code_print(fp)(repr(exc))
         del tb
 
+    def chunk(l, n):
+        value_iterator = iter(l) 
+        return iter(lambda: list(islice(value_iterator, n)), [])
+
+    # greedily concatenate short strings into groups of at most a certain length
+    def chunk_concat(l, n): 
+        current_value = "" 
+        inhabited = False 
+        for text in l:
+            inhabited = True 
+            new_len = len(current_value + text)
+            if new_len > n: 
+                yield current_value 
+                current_value = "" 
+            current_value = current_value + text 
+        if inhabited: 
+            yield current_value 
+
     def format_block(fp):
         text = fp.getvalue()
         return util.discord.format("{!b:py}", text) if len(text) else "\u2705"
+    
+    def short_heuristic(fp): 
+        return len(format_block(fp)) <= 2000
+    
+    def make_file_output(idx, fp): 
+        fp.seek(0)
+        discord_filename = "output{:d}.txt".format(idx)
+        discord_file = discord.File(fp, filename = discord_filename)
+        return discord_file
+
+    message_outputs = chunk_concat((format_block(m) for m in outputs if short_heuristic(m)), 2000) 
+    file_outputs = chunk((make_file_output(*m) for m in enumerate(outputs, start = 1) if not short_heuristic(m[1])), 10)  
```

Comment:
80 cols

Reply:
Should I format it like this or are there too many lines? 

```py
message_outputs = chunk_concat(
    (format_block(m) for m in outputs if short_heuristic(m)), 2000
)
file_outputs = chunk(
    (
        make_file_output(*m)
        for m in enumerate(outputs, start=1)
        if not short_heuristic(m[1])
    ),
    10,
)
```

Reply:
```py
    message_outputs = chunk_concat(
        (format_block(m) for m in outputs if short_heuristic(m)),
        2000)
    file_outputs = chunk(
        (make_file_output(idx, fp) for idx, fp in enumerate(outputs, start = 1)
            if not short_heuristic(idx)),
        10)
```

Reply:
Do you have a formatting tool along with config that you'd recommend, or should I try manually to replicate what's in the code base?